## Valadating the SPIRE red catalogue

![HELP LOGO](https://avatars1.githubusercontent.com/u/7880370?s=100&v=4>)

### three methods of selecting 500um riser galaxies


In [15]:
import numpy as np
%matplotlib inline
from astropy.table import Table, vstack

In [31]:
all_names =  ['GAMA-09','GAMA-12','GAMA-15','HATLAS-NGP','HATLAS-SGP','SSDF',\
              'AKARI-SEP','Bootes','CDFS-SWIRE','COSMOS','EGS',\
              'ELAIS-N1','ELAIS-N2','ELAIS-S1','HDF-N','Lockman-SWIRE','SA13',\
              'SPIRE-NEP','xFLS','XMM-13hr','XMM-LSS','AKARI-NEP']

In [32]:
# Combine all red catalogues
name = 'data/'+all_names[0]+'_SPIRE_D_MF_cat.fits'
cat=Table.read(name)
for j in range(1,np.size(all_names)):
    name = 'data/'+all_names[j]+'_SPIRE_D_MF_cat.fits'
    cat2=Table.read(name)
    cat = vstack([cat, cat2])    

In [33]:
#select 500um risers with S500 > 52 mJy and S/N >3, these limits can easaly be changed depending on the science goal
red_use = (cat['F_RED_MF_SPIRE_250'] < cat['F_RED_MF_SPIRE_350']) & \
          (cat['F_RED_MF_SPIRE_350'] < cat['F_RED_MF_SPIRE_500']) & \
          (cat['F_RED_MF_SPIRE_500'] > 52) & (cat['F_RED_MF_SPIRE_500']/cat['FErr_RED_MF_SPIRE_500'] > 3)
cat_red = cat[red_use]
print('Number of 500um risers with S500 > 52 mJy: ', np.size(cat_red['RA']))


Number of 500um risers with S500 > 52 mJy:  2856


### We find a total of 2856 500um risers with S500 > 53 mJy over an area of 900 deg2.
### Asboth et al. 2016 found 477 of such SPIRE detected galaxies in the HELMS field. 

In [49]:
# Combining all blind detectet catalogues
name = '../dmu22_'+all_names[0]+'/data/dmu22_XID+SPIRE_'+all_names[0]+'_BLIND_Matched_MF.fits'
cat_XID=Table.read(name)
for j in range(1,np.size(all_names)):
    name = '../dmu22_'+all_names[j]+'/data/dmu22_XID+SPIRE_'+all_names[j]+'_BLIND_Matched_MF.fits'
    cat2=Table.read(name)
    print(cat2['F_BLIND_MF_SPIRE_250'].unit,all_names[j])
    cat_XID = vstack([cat_XID, cat2])   

mJy GAMA-12


mJy GAMA-15


mJy HATLAS-NGP


mJy HATLAS-SGP


mJy SSDF


mJy AKARI-SEP


mJy Bootes


mJy CDFS-SWIRE


mJy COSMOS


mJy EGS


mJy ELAIS-N1


mJy ELAIS-N2


mJy ELAIS-S1


mJy HDF-N


mJy Lockman-SWIRE


mJy SA13


mJy SPIRE-NEP


mJy xFLS


mJy XMM-13hr


Jy XMM-LSS


mJy AKARI-NEP


In [50]:
# selectng 500um risers based on the MF flux density
red_use_MF = (cat_XID['F_BLIND_MF_SPIRE_250'] < cat_XID['F_BLIND_MF_SPIRE_350']) & \
          (cat_XID['F_BLIND_MF_SPIRE_350'] < cat_XID['F_BLIND_MF_SPIRE_500']) & \
          (cat_XID['F_BLIND_MF_SPIRE_500'] > 52) & \
          (cat_XID['F_BLIND_MF_SPIRE_500']/cat_XID['FErr_BLIND_MF_SPIRE_500'] > 3)

cat_red_MF = cat_XID[red_use_MF]
print('Number of 500um risers with S500 > 52 mJy: ', np.size(cat_red_MF['RA']))

# selectng 500um risers based on the BLIND XID+ flux density
err = cat_XID['F_SPIRE_500'] - cat_XID['FErr_SPIRE_500_l']
red_use = (cat_XID['F_SPIRE_250'] < cat_XID['F_SPIRE_350']) & \
          (cat_XID['F_SPIRE_350'] < cat_XID['F_SPIRE_500']) & \
          (cat_XID['F_SPIRE_500'] > 52) & (cat_XID['F_SPIRE_500']/err > 3)
cat_red_XID = cat_XID[red_use]
print('Number of 500um risers with S500 > 52 mJy: ', np.size(cat_red_XID['RA']))


Number of 500um risers with S500 > 52 mJy:  2728
Number of 500um risers with S500 > 52 mJy:  1453


/Users/Steven/anaconda/envs/herschelhelp_internal/lib/python3.6/site-packages/astropy/table/column.py:965: RuntimeWarning: invalid value encountered in less
  return getattr(self.data, op)(other)
/Users/Steven/anaconda/envs/herschelhelp_internal/lib/python3.6/site-packages/astropy/table/column.py:965: RuntimeWarning: invalid value encountered in greater
  return getattr(self.data, op)(other)


![HELP LOGO](https://avatars1.githubusercontent.com/u/7880370?s=75&v=4)

**Authors**: Steven Duivenvoorden 

For a full description of the database and how it is organised in to `dmu_products` please the top level [readme](../readme.md).

The Herschel Extragalactic Legacy Project, ([HELP](http://herschel.sussex.ac.uk/)), is a [European Commission Research Executive Agency](https://ec.europa.eu/info/departments/research-executive-agency_en)
funded project under the SP1-Cooperation, Collaborative project, Small or medium-scale focused research project, FP7-SPACE-2013-1 scheme, Grant Agreement Number 607254.

[Acknowledgements](http://herschel.sussex.ac.uk/acknowledgements)